script.js

```diff
@@ -1,66 +1,462 @@
+			"<button id='delete-" + searchResult[eachObj].id + "' class='delete_button'>" + "Delete" + "</button>" +
+			"<button id='cancel-" + searchResult[eachObj].id + "' class='cancel_button'>" + "Cancel" + "</button>" +
+			"</div>" +
+			"<div class='buttons'>" +
+			"<button id='edit-" + searchResult[eachObj].id + "' class='edit_button'><img class='editButtonImage' id='editimage" + searchResult[eachObj].id + "' src='./images/edit.png'></button>" +
+			"<button id='remove-" + searchResult[eachObj].id + "' class='remove_button'><img id='removeimage" + searchResult[eachObj].id + "' src='./images/icons8-trash-can-50.png'></button>" +
+			"</div></li>";
+		_("printing_list").insertAdjacentHTML('afterbegin', result);
+
+		let menu = _("menu-" + searchResult[eachObj].id);
+		let dotmenu = _("dotmenu-" + searchResult[eachObj].id);
+		function displayButtons(e) {
+			e.stopPropagation();
+			_("overlay").classList.add("flexDisplay");
+			_("floatDiv").classList.add("floatDiv");
+			_("floatButton").classList.add("noshow");
+		}
+		menu.addEventListener("click", displayButtons);
+		dotmenu.addEventListener("click", displayButtons);
+		let checkBox = _("check-" + searchResult[eachObj].id)
+		let description = _("desc-" + searchResult[eachObj].id)
+
+		//cross-out on clicking description or checkbox
+		checkBox.addEventListener("change", (e) => {
+			let targetId = e.target.id.replace("check-", "")
+			if (list.find(element => {
+				return element.id.toString() === targetId
+			}).editMode === false) {
+				toggleCheckedClass(targetId);
+			}
+		})
 
+		description.addEventListener("click", function (e) {
+			let targetId = e.target.id.replace("desc-", "")
+			if (list.find(element => {
+				return element.id.toString() === targetId
+			}).editMode === false) {
+				checkBox.checked = checkBox.checked ? false : true;
+				toggleCheckedClass(targetId);
+			}
+		})
+
+		//Edit button functionalities
+		let editItem = _("edit-" + searchResult[eachObj].id);
+		let editimage = _("editimage" + searchResult[eachObj].id)
+		function editContent(e) {
+			e.stopPropagation();
+			let targetElem = e.target.id.replace("editimage", "").replace("edit-", "").replace("description", "").replace("desc-", "")
+			if (list.find(elem => elem.id.toString() === targetElem).editMode === false) {
+				if (_('check-' + targetElem).checked === true) {
+					_('desc-' + targetElem).classList.remove("checkboxChecked");
+				}
+				list.find(elem => elem.id.toString() === targetElem).editMode = true;
+				_('desc-' + targetElem).contentEditable = true;
+				_('check-' + targetElem).setAttribute("disabled", "disabled");
+				editimage.classList.add("saveButtonImage");
+				editimage.classList.remove("editButtonImage");
+				editimage.setAttribute("src", './images/save.png')
+				_('desc-' + targetElem).focus();
+				let placeOfEdit = _('desc-' + targetElem)
+				function placeCaretAtEnd(el) {
+					el.focus();
```

Comment:
We already have this function defined above. Move it to another file as mentioned earlier.